include/toolchain/gcc.h

```diff
@@ -27,7 +27,17 @@
 #define FUNC_ALIAS(real_func, new_alias, return_type) \
 	return_type new_alias() ALIAS_OF(real_func)
 
+#if defined(CONFIG_ARCH_POSIX)
+/*let's not segfault if this were to happen for some reason*/
+#define CODE_UNREACHABLE \
+{\
+	void exit(int status); \
+	exit(1);\
```

Comment:
would a call to k_panic() work? would help users comprehend that something bad happened and where it occurred.

Reply:
alternatively, put an __ASSERT() here instead?

Reply:
Uhm, for ASSERT that assumes misc/__assert.h has been included at that point.
regarding k_panic() also needs kernel.h.
What about just replacing the exit call with a ps_print_error_and_exit("CODE_UNREACHABLE reached from %s:%d", __FILE__ , __LINE__);
+ #include "posix_soc_if.h" at the top?